server/src/schema.ts

```diff
@@ -6,190 +6,285 @@ import utils from './utils'
+    //   return { users, count, take }
+    // },
   },
   Mutation: {
-    deleteUser: (parent, args, ctx: Context) => {
-      return ctx.prisma.user.delete({
-        where: { id: args.userId },
-      })
-    },
-    updateUser: async (parent, args, ctx: Context) => {
-      const userId = utils.getUserId(ctx)
-      const me = await ctx.prisma.user.findUnique({ where: { id: userId } })
-      if (!me) throw new Error('Not Auth')
+    signupUser: async (parent, args, ctx: Context) => {
+      const { email, password } = args
+      const hashedPassword = await bcrypt.hash(password, 10)
 
-      return ctx.prisma.user.update({
-        where: { id: args.userId },
-        data: {
-          name: args.data.name,
-          role: me.role === 'ADMIN' ? args.data.role : undefined,
-        },
-      })
+      try {
+        // @ts-ignore
+        const user = await ctx.prisma.user.create({
+          data: {
+            email,
+            password: hashedPassword,
+          },
+        })
+        const payload = {
+          userId: user.id,
+          userEmail: user.email,
+        }
+        const token = jwt.sign(payload, config.APP_SECRET, { expiresIn: '2d' })
+        return {
+          user,
+          token,
+        }
+      } catch (error) {
+        console.error(error)
+        throw new AuthenticationError(error.message || 'Kek')
+      }
     },
+    addSourceToFeed: async (parent, args, ctx: Context) => {
+      const { feedId, sourceId } = args
 
-    forgetPassword: async (parent, args, ctx: Context) => {
-      let user = await ctx.prisma.user.findUnique({
-        where: {
-          email: args.email,
-        },
-      })
+      try {
+        const relation = await ctx.prisma.sourceFeedRelation.findFirst({
+          where: {
+            feedId,
+            sourceId,
+          },
+        })
 
-      if (!user) {
-        throw new Error('Email unknown')
-      }
-      const resetPasswordToken = crypto.randomBytes(64).toString('hex')
-      user = await ctx.prisma.user.update({
-        where: {
-          id: user.id,
-        },
-        data: {
-          resetPasswordToken,
-          dateResetPasswordRequest: new Date(),
-        },
-      })
-      email.sendForgetPassword(ctx, user)
+        if (!relation) {
+          await ctx.prisma.sourceFeedRelation.create({
+            data: {
+              feedId,
+              sourceId,
+            },
+          })
 
-      return true
-    },
-    resetPassword: async (parent, args, ctx: Context) => {
-      if (!args.resetPasswordToken) {
-        throw new Error('Error. No token')
-      }
-      let user = await ctx.prisma.user.findFirst({
-        where: {
-          resetPasswordToken: args.resetPasswordToken,
-        },
-      })
-
-      if (!user) {
-        throw new Error('Link not valide!')
+          return {
+            feedId,
+          }
+        }
+      } catch (error) {
+        console.error(error)
```

Comment:
hmm...you should almost never be wrapping whole function bodies in try catch like this.